docs/library/machine.I2S.rst

```diff
@@ -0,0 +1,157 @@
+                    
+3 modes of operation are supported:
+ - blocking 
+ - non-blocking 
+ - uasyncio
+  
+blocking::
+ 
+   num_written = audio_out.write(buf) # blocks until buf emptied
+
+   num_read = audio_in.readinto(buf) # blocks until buf filled
+   
+non-blocking::
+ 
+   audio_out.callback(i2s_callback)    # i2s_callback is called when buf is emptied
+   num_written = audio_out.write(buf)  # returns immediately
+        
+   audio_in.callback(i2s_callback)     # i2s_callback is called when buf is filled
+   num_read = audio_in.readinto(buf)   # returns immediately    
+ 
+uasyncio::
+ 
+   swriter = uasyncio.StreamWriter(audio_out)
+   swriter.write(buf)
+   await swriter.drain()
+   
+   sreader = uasyncio.StreamReader(audio_in)
+   num_read = await sreader.readinto(buf)
+ 
+Constructor
+-----------
+
+.. class:: I2S(id, *, sck, ws, sd, mode, bits, format, rate, bufferlen)
+
+   Construct an I2S object of the given id:
+   
+   - ``id`` identifies a particular I2S bus.  
+
+   ``id`` is board and port specific:
+ 
+     - PYBv1.0/v1.1: has one I2S bus with id=2.
+     - PYBD-SFxW: has two I2S busses with id=1 and id=2. 
+     - ESP32: has two I2S busses with id=0 and id=1. 
+   
+   Keyword-only parameters that are supported on all ports:
+    
+     - ``sck`` is a pin object for the serial clock line
+     - ``ws`` is a pin object for the word select line
+     - ``sd`` is a pin object for the serial data line
+     - ``mode`` specifies receive or transmit
+     - ``bits`` specifies sample size (bits), 16 or 32
+     - ``format`` specifies channel format, STEREO or MONO
+     - ``rate`` specifies audio sampling rate (samples/s)
+     - ``bufferlen`` specifies internal buffer length (bytes)
+     
+   For all ports, DMA runs continuously in the background and allows user applications to perform other operations while 
+   sample data is transfered between the internal buffer and the I2S peripheral unit. 
+   Increasing the size of the internal buffer has the potential to increase the time that user applications can perform non-I2S operations
+   before underflow (e.g. ``write`` method) or overflow (e.g. ``readinto`` method).
+
+Methods
+-------
+
+.. method:: I2S.init(sck, ...)
+
+  see Constructor for argument descriptions
+     
+.. method:: I2S.deinit()
+
+  Deinitialize the I2S bus
+  
+.. method::  I2S.readinto(buf)
+
+  Read audio samples into the buffer specified by ``buf``.  ``buf`` must support the buffer protocol, such as bytearray or array. 
+  "buf" byte ordering is little-endian.  For Stereo format, left channel sample precedes right channel sample. 
+  Returns number of bytes read 
+  
+.. method::  I2S.write(buf)
+
+  Write audio samples contained in ``buf``. ``buf`` must support the buffer protocol, such as bytearray or array.
+  "buf" byte ordering is little-endian.  For Stereo format, left channel sample precedes right channel sample.
+  Returns number of bytes written 
+  
+.. method::  I2S.callback(handler)
```

Comment:
To match other classes (eg Pin, BLE), I'd suggest renaming this method to `I2S.irq(handler)`.

Reply:
Thanks for flagging that.  I tried to keep consistency with other modules, but I missed that one.  I'll update the PR.

Reply:
I added a new PR commit that bring consistency to the callback naming.